packages/ui/report.api.md

```diff
@@ -1769,35 +1698,48 @@ export interface TextFieldProps
 
 // @public (undocumented)
 export type TextOwnProps = {
-  as?: 'p' | 'span' | 'label';
-  variant?:
-    | 'subtitle'
-    | 'body'
-    | 'caption'
+  as?:
+    | 'h1'
+    | 'h2'
+    | 'h3'
+    | 'h4'
+    | 'h5'
+    | 'h6'
+    | 'p'
+    | 'span'
     | 'label'
-    | Partial<Record<Breakpoint, 'subtitle' | 'body' | 'caption' | 'label'>>;
-  weight?: 'regular' | 'bold' | Partial<Record<Breakpoint, 'regular' | 'bold'>>;
+    | 'div'
+    | 'strong'
+    | 'em'
+    | 'small'
+    | 'legend';
+  variant?: TextVariants | Partial<Record<Breakpoint, TextVariants>>;
+  weight?: TextWeights | Partial<Record<Breakpoint, TextWeights>>;
   color?:
-    | 'primary'
-    | 'secondary'
-    | 'danger'
-    | 'warning'
-    | 'success'
-    | Partial<
-        Record<
-          Breakpoint,
-          'primary' | 'secondary' | 'danger' | 'warning' | 'success'
-        >
-      >;
+    | TextColors
+    | TextColorStatus
+    | Partial<Record<Breakpoint, TextColors | TextColorStatus>>;
   truncate?: boolean;
-  className?: string;
-  style?: React.CSSProperties;
 };
 
 // @public (undocumented)
-export type TextProps<T extends ElementType = 'p'> = TextOwnProps &
+export type TextProps<T extends ElementType = 'span'> = TextOwnProps &
   Omit<ComponentPropsWithRef<T>, keyof TextOwnProps>;
 
+// @public (undocumented)
+export type TextVariants =
```

Comment:
Thoughts on this vs `variant: title | body` + `size: x-small | small | medium | large`?

Reply:
Interesting. This is a nice pattern for sure to have the same values for both `title` and `body` but we might bring a bigger size later on or another variant for uppercase small types that would break this pattern. Perhaps keeping it all on the same variant is better future proof. WDYT?